noir-projects/aztec-nr/aztec/src/state_vars/shared_mutable/shared_mutable.nr

```diff
@@ -225,6 +225,34 @@ impl<T, INITIAL_DELAY> SharedMutable<T, INITIAL_DELAY, &mut PrivateContext> {
     }
 }
 
+impl<T, INITIAL_DELAY> SharedMutable<T, INITIAL_DELAY, UnconstrainedContext> where T: ToField + FromField + Eq {
+    pub fn get_current_value_in_unconstrained(self) -> T {
+        let block_number = self.context.block_number() as u32;
+        self.read_value_change().get_current_at(block_number)
+    }
+
+    pub fn get_current_delay_in_unconstrained(self) -> u32 {
+        let block_number = self.context.block_number() as u32;
+        self.read_delay_change().get_current(block_number)
+    }
+
+    pub fn get_scheduled_value_in_unconstrained(self) -> (T, u32) {
+        self.read_value_change().get_scheduled()
+    }
+
+    pub fn get_scheduled_delay_in_unconstrained(self) -> (u32, u32) {
+        self.read_delay_change().get_scheduled()
+    }
+
+    fn read_value_change(self) -> ScheduledValueChange<T> {
+        self.context.storage_read(self.get_value_change_storage_slot())
+    }
+
+    fn read_delay_change(self) -> ScheduledDelayChange<INITIAL_DELAY> {
+        self.context.storage_read(self.get_delay_change_storage_slot())
+    }
+}
```

Comment:
These fns should be `unconstrained`, both to flag that the execution is not constrained, and to avoid issues that might arise if we end up passing mut& here.

In the future this will be more obvious because https://github.com/noir-lang/noir/issues/4442 would require that you place this in an `unsafe` block, and you'd go 'oh i don't actually want to do that'.